### PR TITLE
[Core] cleanup connectivity-preserve-modeler

### PR DESCRIPTION
**Pull request by philbucher**
cleaning up a bit before actually changing sth to make the review easier

### CODE REVIEW

File: kratos/sources/connectivity_preserve_modeler.cpp [hunk 1]
```diff
@@ -181,39 +157,29 @@ void ConnectivityPreserveModeler::DuplicateCommunicatorData(
...
         pDestinationComm->pGhostMesh(i)->SetNodes( rReferenceComm.pGhostMesh(i)->pNodes() );
     }
 
-    // This is a dirty hack to detect if the communicator is a Communicator or an MPICommunicator
-    // Note that downcasting would not work here because MPICommunicator is not compiled in non-MPI builds
-    bool is_mpi = ( rOriginModelPart.pElements() == rReferenceComm.LocalMesh().pElements() );
-
-    if (is_mpi)
-    {
+    if (DataCommunicator::GetDefault().IsDistributed()) {
```
Comment (philbucher) [inline kratos/sources/connectivity_preserve_modeler.cpp]: @jcotela this is correct right?

**Review by loumalouomega — DISMISSED** [review]
OK from my side

**Comment by philbucher** [comment]
hm some FSIApp-test is failing, not sure if related to this PR or not

@rubenzorrilla I don't think you use the ConnectivityPreserveModeler in FSI do you? At least I didn't find any usages (Also neither Fluid nor Structure use it afaik)

**Comment by philbucher** [comment]
> hm some FSIApp-test is failing, not sure if related to this PR or not
> 
> @rubenzorrilla I don't think you use the ConnectivityPreserveModeler in FSI do you? At least I didn't find any usages (Also neither Fluid nor Structure use it afaik)

hm false alarm, see #4256 , does not seem to be related to this PR

File: kratos/sources/connectivity_preserve_modeler.cpp (hunk 1 above)
Comment (philbucher) [reply kratos/sources/connectivity_preserve_modeler.cpp]: ping @jcotela

**Comment by philbucher** [comment]
any other comments?

**Comment by rubenzorrilla** [comment]
Sorry for the late reply, no #4256 shouldn't be related to the connectivity preserve modeler.

**Comment by philbucher** [comment]
ping @jcotela

**Comment by philbucher** [comment]
ping @jcotela

File: kratos/sources/connectivity_preserve_modeler.cpp [hunk 2]
```diff
@@ -122,17 +102,15 @@ void ConnectivityPreserveModeler::CopyCommonData(
     rDestinationModelPart.AddNodes(rOriginModelPart.NodesBegin(), rOriginModelPart.NodesEnd());
 }
 
-
 void ConnectivityPreserveModeler::DuplicateElements(
-    ModelPart &rOriginModelPart,
-    ModelPart &rDestinationModelPart,
-    Element const &rReferenceElement)
+    ModelPart& rOriginModelPart,
```
Comment (RiccardoRossi) [inline kratos/sources/connectivity_preserve_modeler.cpp:106]: can't we make const the origin modelpart?

File: kratos/sources/connectivity_preserve_modeler.cpp [hunk 3]
```diff
@@ -144,17 +122,15 @@ void ConnectivityPreserveModeler::DuplicateElements(
     rDestinationModelPart.AddElements(temp_elements.begin(), temp_elements.end());
 }
 
-
 void ConnectivityPreserveModeler::DuplicateConditions(
-        ModelPart &rOriginModelPart,
-        ModelPart &rDestinationModelPart,
-        Condition const &rReferenceBoundaryCondition)
+    ModelPart& rOriginModelPart,
```
Comment (RiccardoRossi) [inline kratos/sources/connectivity_preserve_modeler.cpp:126]: same

**Review by RiccardoRossi — COMMENTED** [review]
minor comment

File: kratos/sources/connectivity_preserve_modeler.cpp (hunk 2 above)
Comment (philbucher) [reply kratos/sources/connectivity_preserve_modeler.cpp:106]: I tried, but not possible bcs some methods are not const, e.g. [those](https://github.com/KratosMultiphysics/Kratos/blob/d5b63540b10a651f110906e0967a4c5707dbd454/kratos/sources/connectivity_preserve_modeler.cpp#L97-L99)

**Comment by RiccardoRossi** [comment]
ok then, i think i can approve this

**Review by RiccardoRossi — APPROVED** [review]

**Comment by philbucher** [comment]
thx

**Pull request merged**